conans/model/conan_file.py

```diff
@@ -150,14 +151,38 @@ def __init__(self, output, runner, display_name="", user=None, channel=None):
         self._conan_using_build_profile = False
         self._conan_requester = None
 
-        self.layout = Layout()
         self.buildenv_info = Environment()
         self.runenv_info = Environment()
         self._conan_buildenv = None  # The profile buildenv, will be assigned initialize()
         self._conan_node = None  # access to container Node object, to access info, context, deps...
         self.virtualenv = True  # Set to false to opt-out automatic usage of VirtualEnv
 
-        self._conan_new_cpp_info = None  # Will be calculated lazy in the getter
+        self._conan_new_cpp_info = None   # Will be calculated lazy in the getter
+
+        # layout() method related variables:
+        self.folders = Folders()
+        self.patterns = Patterns()
+        self.infos = Infos()
+
+        self.patterns.source.include = ["*.h", "*.hpp", "*.hxx"]
+        self.patterns.source.lib = []
+        self.patterns.source.bin = []
+
+        self.patterns.build.include = ["*.h", "*.hpp", "*.hxx"]
+        self.patterns.build.lib = ["*.so", "*.so.*", "*.a", "*.lib", "*.dylib"]
+        self.patterns.build.bin = ["*.exe", "*.dll"]
+
+        self.infos.package.includedirs = ["include"]
+        self.infos.package.libdirs = ["lib"]
+        self.infos.package.bindirs = ["bin"]
+        self.infos.package.resdirs = ["res"]
+        self.infos.package.builddirs = [""]
+        self.infos.package.frameworkdirs = ["Frameworks"]
+
+        # Init all fields with None so we can certainly know when a field has been declared
+        # to be able to merge them to the package one
+        self.infos.source = NewCppInfo()
```

Comment:
Isn't this the ``Infos()`` default already?

Reply:
yes, this is no longer necessary because I simplified the None stuff.